src/examples/src/widgets/form/Basic.tsx

```diff
@@ -0,0 +1,165 @@
+import { create, tsx } from '@dojo/framework/core/vdom';
+import { icache } from '@dojo/framework/core/middleware/icache';
+
+import Button from '@dojo/widgets/button';
+import TextInput from '@dojo/widgets/text-input';
+import Form from '@dojo/widgets/form';
+
+const factory = create({ icache });
+
+interface Fields {
+	firstName: string;
+	middleName?: string;
+	lastName: string;
+	email?: string;
+}
+
+const App = factory(function({ middleware: { icache } }) {
+	const results = icache.get<Partial<Fields>>('basic');
+	const onValueResults = icache.get<Partial<Fields>>('basicOnValue');
+
+	return (
+		<virtual>
+			<Form
+				initialValue={{
+					firstName: 'Billy'
+				}}
+				onSubmit={(values) => icache.set('basic', values)}
+				onValue={(values) =>
+					icache.set('basicOnValue', { ...icache.get('basicOnValue'), ...values })
+				}
+			>
+				{({ values, required, valid, form, disabled }) => {
+					const { get, set } = values<Fields>();
+
+					required('firstName', true);
+					required('lastName', true);
+
+					return [
+						<TextInput
+							key="firstName"
+							label="First Name"
+							placeholder="Enter first name (must be Billy)"
+							pattern="Billy"
+							required={true}
+							value={get('firstName')}
```

Comment:
could this work as `field('firstName').value`, `field('firstName').required(true)` etc perhaps ? then this would line up to checkbox group etc better and make the number of functions passed into the renderer more succinct.